### PR TITLE
Bah 987 newbranch 6 dec 2020

### PR DESCRIPTION
**Pull request by buvaneswari-arun**

### CODE REVIEW

File: ui/app/clinical/consultation/controllers/dispositionController.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 'use strict';
 
 angular.module('bahmni.clinical')
-    .controller('DispositionController', ['$scope', '$q', 'dispositionService', 'retrospectiveEntryService', 'spinner', function ($scope, $q, dispositionService, retrospectiveEntryService, spinner) {
+    .controller('DispositionController', ['$scope', '$q', 'dispositionService', 'appService', 'retrospectiveEntryService', 'spinner', '$rootScope', '$translate', function ($scope, $q, dispositionService, appService, retrospectiveEntryService, spinner, $rootScope, $translate) {
         var consultation = $scope.consultation;
         var allDispositions = [];
-
+        $scope.ModuleName = appService.getAppDescriptor().getConfigValue('disposition');
```
Comment (binduak) [inline ui/app/clinical/consultation/controllers/dispositionController.js:7]: @buvaneswari-arun it should be "moduleName". not "ModuleName". please correct that

File: ui/app/clinical/displaycontrols/patientContext/directives/patientContext.js [hunk 2]
```diff
@@ -44,6 +44,18 @@ angular.module('bahmni.clinical')
             });
         };
 
+        var translateAttributes = function (attributes) {
+            _.forEach(attributes, function (attribute, key) {
+                var keyPrefix = "";
+                var keyName = key.toUpperCase().replace(/\s\s+/g, ' ').replace(/[^a-zA-Z0-9 _]/g, "").trim().replace(/ /g, "_");
```
Comment (binduak) [inline ui/app/clinical/displaycontrols/patientContext/directives/patientContext.js:50]: @buvaneswari-arun Can we move keyName regression to a util as its being used in multiple places

File: ui/app/clinical/displaycontrols/treatmentData/directives/treatmentData.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ angular.module('bahmni.clinical')
 
                 var startDate = null, endDate = null, getEffectiveOrdersOnly = false;
                 if (programConfig.showDetailsWithinDateRange) {
-                    startDate = $stateParams.dateEnrolled;
+                    startDate = $stateParamqs.dateEnrolled;
```
Comment (binduak) [inline ui/app/clinical/displaycontrols/treatmentData/directives/treatmentData.js:24]: @buvaneswari-arun can you check is this `stateParamqs` or `stateParams` ?

File: ui/app/common/displaycontrols/observation/directives/bahmniObservation.js [hunk 4]
```diff
@@ -94,7 +94,15 @@ angular.module('bahmni.common.displaycontrol.observation')
                         }
                     }
                 };
-
+                $scope.translateAttributeName = function (attribute) {
+                    var keyName = attribute.toUpperCase().replace(/\s\s+/g, ' ').replace(/[^a-zA-Z0-9 _]/g, "").trim().replace(/ /g, "_");
```
Comment (binduak) [inline ui/app/common/displaycontrols/observation/directives/bahmniObservation.js:98]: @buvaneswari-arun same comment as above

File: ui/app/common/displaycontrols/programs/directives/programs.js [hunk 5]
```diff
@@ -55,6 +60,19 @@ angular.module('bahmni.common.displaycontrol.programs')
...
+                        attribute.description = attribute.display;
+                    }
+                    var keyName = attribute.description.toUpperCase().replace(/[^a-zA-Z0-9]/g, "").replace(/ /g, "_").replace(/-/g, "_");
+                    var translationKey = keyPrefix + '_' + keyName;
+                    var translation = $translate.instant(translationKey);
+                    if (translation == translationKey) {
+                        return attribute.description;
+                    }
+                    return translation;
+                };
```
Comment (binduak) [inline ui/app/common/displaycontrols/programs/directives/programs.js:75]: @buvaneswari-arun I see the similar method has been used in multiple places. Can we move this to a util file

File: ui/app/registration/controllers/createPatientController.js [hunk 6]
```diff
@@ -11,11 +11,24 @@ angular.module('bahmni.registration')
...
-
+            $scope.translateAttributes = function (attribute) {
+                if ($scope.ModuleName == null) {
+                    var keyPrefix = "REGISTRATION";
+                } else {
+                    var keyPrefix = $scope.ModuleName;
+                }
+                var keyName = attribute.toUpperCase().replace(/\s\s+/g, ' ').replace(/[^a-zA-Z0-9 _]/g, "").trim().replace(/ /g, "_");
+                var translationKey = keyPrefix + "_" + keyName;
+                var translation = $translate.instant(translationKey);
```
Comment (binduak) [inline ui/app/registration/controllers/createPatientController.js:26]: @buvaneswari-arun same comment as above

File: ui/app/registration/controllers/createPatientController.js [hunk 7]
```diff
@@ -11,11 +11,24 @@ angular.module('bahmni.registration')
             $scope.disablePhotoCapture = appService.getAppDescriptor().getConfigValue("disablePhotoCapture");
             $scope.showEnterID = configValueForEnterId === null ? true : configValueForEnterId;
             $scope.today = Bahmni.Common.Util.DateTimeFormatter.getDateWithoutTime(dateUtil.now());
-
+            $scope.ModuleName = appService.getAppDescriptor().getConfigValue('registrationModuleName');
```
Comment (binduak) [inline ui/app/registration/controllers/createPatientController.js:14]: @buvaneswari-arun same comment as above

File: ui/app/registration/controllers/patientCommonController.js [hunk 8]
```diff
@@ -17,7 +17,7 @@ angular.module('bahmni.registration')
             $scope.readOnlyExtraIdentifiers = appService.getAppDescriptor().getConfigValue("readOnlyExtraIdentifiers");
             $scope.showSaveConfirmDialogConfig = appService.getAppDescriptor().getConfigValue("showSaveConfirmDialog");
             $scope.showSaveAndContinueButton = false;
-
+            $scope.ModuleName = appService.getAppDescriptor().getConfigValue('registrationModuleName');
```
Comment (binduak) [inline ui/app/registration/controllers/patientCommonController.js:20]: @buvaneswari-arun same comment as above

File: ui/app/registration/controllers/patientCommonController.js [hunk 9]
```diff
@@ -42,7 +42,20 @@ angular.module('bahmni.registration')
...
+                    var keyPrefix = $scope.ModuleName;
+                }
+                var keyName = attribute.toUpperCase().replace(/\s\s+/g, ' ').replace(/[^a-zA-Z0-9 _]/g, "").trim().replace(/ /g, "_");
+                var translationKey = keyPrefix + "_" + keyName;
+                var translation = $translate.instant(translationKey);
+                if (translation != translationKey) {
+                    attribute = translation;
+                }
+                return attribute;
+            };
```
Comment (binduak) [inline ui/app/registration/controllers/patientCommonController.js:58]: @buvaneswari-arun same comment as above

File: ui/app/registration/directives/addressFields.js [hunk 10]
```diff
@@ -36,7 +40,20 @@ angular.module('bahmni.registration')
...
+                var keyPrefix = moduleName && modulePrefixMap[moduleName] ? modulePrefixMap[moduleName] : '';
+            }
+            var keyName = attribute.toUpperCase().replace(/\s\s+/g, ' ').replace(/[^a-zA-Z0-9 _]/g, "").trim().replace(/ /g, "_");
+            var translationKey = keyPrefix + "_" + keyName;
+            var translation = $translate.instant(translationKey);
+            if (translation != translationKey) {
+                attribute = translation;
+            }
+            return attribute;
+        };
```
Comment (binduak) [inline ui/app/registration/directives/addressFields.js:56]: @buvaneswari-arun same comment as above

File: ui/app/registration/directives/topDownAddressFields.js [hunk 11]
```diff
@@ -44,7 +48,20 @@ angular.module('bahmni.registration')
...
+                var keyPrefix = moduleName && modulePrefixMap[moduleName] ? modulePrefixMap[moduleName] : '';
+            }
+            var keyName = attribute.toUpperCase().replace(/\s\s+/g, ' ').replace(/[^a-zA-Z0-9 _]/g, "").trim().replace(/ /g, "_");
+            var translationKey = keyPrefix + "_" + keyName;
+            var translation = $translate.instant(translationKey);
+            if (translation != translationKey) {
+                attribute = translation;
+            }
+            return attribute;
+        };
```
Comment (binduak) [inline ui/app/registration/directives/topDownAddressFields.js:64]: @buvaneswari-arun same comment as above

Comment (binduak) [reply ui/app/registration/directives/topDownAddressFields.js:64]: @buvaneswari-arun As this method is not private and being used in multiple places, please add test cases for this piece of code

**Review by binduak — CHANGES_REQUESTED** [review]
@buvaneswari-arun Please check the mentioned review comments

File: ui/app/clinical/displaycontrols/treatmentData/directives/treatmentData.js (hunk 3 above)
Comment (buvaneswari-arun) [reply ui/app/clinical/displaycontrols/treatmentData/directives/treatmentData.js]: Modified

File: ui/app/common/displaycontrols/observation/directives/bahmniObservation.js (hunk 4 above)
Comment (buvaneswari-arun) [reply ui/app/common/displaycontrols/observation/directives/bahmniObservation.js:98]: Moved to a separate Util method

File: ui/app/common/displaycontrols/programs/directives/programs.js (hunk 5 above)
Comment (buvaneswari-arun) [reply ui/app/common/displaycontrols/programs/directives/programs.js]: Done. TranslationUtil.js is newly added and this method is moved there.

File: ui/app/registration/directives/topDownAddressFields.js (hunk 11 above)
Comment (buvaneswari-arun) [reply ui/app/registration/directives/topDownAddressFields.js]: Testcases will be taken up as a part of second cut.

**Review by binduak — APPROVED** [review]

**Pull request merged**